Sources/BraintreePaymentFlow/BTLocalPaymentClient.swift

```diff
@@ -0,0 +1,313 @@
+import Foundation
+import AuthenticationServices
+
+#if canImport(BraintreeCore)
+import BraintreeCore
+#endif
+
+#if canImport(BraintreeDataCollector)
+import BraintreeDataCollector
+#endif
+
+@objcMembers public class BTLocalPaymentClient: NSObject {
+    
+    // MARK: - Internal Properties
+    
+    var authenticationSession: ASWebAuthenticationSession?
+
+    /// exposed for testing
+    var merchantCompletion: ((BTLocalPaymentResult?, Error?) -> Void)? = nil
+    
+    // MARK: - Private Properties
+    
+    private let apiClient: BTAPIClient
+    private var request: BTLocalPaymentRequest?
+
+    // MARK: - Initializer
+
+    /// Initialize a new `BTLocalPaymentClient` instance.
+    /// - Parameter apiClient: An API client
+    @objc(initWithAPIClient:)
+    public init(apiClient: BTAPIClient) {
+        self.apiClient = apiClient
+    }
+
+    // MARK: - Public Methods
+
+    /// Starts a payment flow using a `BTLocalPaymentRequest`
+    /// - Parameters:
+    ///   - request: A `BTLocalPaymentRequest` request.
+    ///   - completion: This completion will be invoked exactly once when the payment flow is complete or an error occurs.
+    public func startPaymentFlow(_ request: BTLocalPaymentRequest, completion: @escaping (BTLocalPaymentResult?, Error?) -> Void) {
+        apiClient.sendAnalyticsEvent("ios.\(request.paymentType ?? "").local-payment.start-payment.selected")
+
+        self.request = request
+        self.merchantCompletion = completion
+
+        apiClient.fetchOrReturnRemoteConfiguration { configuration, error in
+            if let error {
+                self.apiClient.sendAnalyticsEvent("ios.\(request.paymentType ?? "").local-payment.start-payment.failed")
+                completion(nil, error)
+                return
+            }
+
+            let dataCollector = BTDataCollector(apiClient: self.apiClient)
+            request.correlationID = dataCollector.clientMetadataID(nil)
+
+            guard let configuration else {
+                self.apiClient.sendAnalyticsEvent("ios.\(request.paymentType ?? "").local-payment.start-payment.failed")
+                completion(nil, BTLocalPaymentError.fetchConfigurationFailed)
+                return
+            }
+
+            if !configuration.isLocalPaymentEnabled {
+                NSLog("%@ Enable PayPal for this merchant in the Braintree Control Panel to use Local Payments.", BTLogLevelDescription.string(for: .critical))
```

Comment:
Minor version TODO - audit our use of logging throughout the SDK. It's inconsistent b/w modules

Reply:
Totally agree. There are a ton of alternatives to `NSLog` that we can explore as well. We should likely follow Apple's recommended path and use their logger: https://developer.apple.com/documentation/os/logging. Added this to our minor version doc to come back to.

Reply:
Awesome - was just about to add this to our V6 minor version plan, but you BEAT ME TO IT!